### ISSUE TITLE
Progression of Luxury Autos to a Business

### ISSUE THREAD

**Issue by claymstehling** [InZidiuZ/legacy-assets-issue-tracker#487]
**Name:**
claystehling#4598

**Summary:**
I would like to request that Luxury Autos become a business in the city rather than operating on a business license, with that there are some changes that I wanted to request along side with it. 

**Link:**
Here are screenshots detailing where the requested items would go. 

https://cdn.discordapp.com/attachments/630409333636595732/882003277858869258/locks1.jpg
https://cdn.discordapp.com/attachments/630409333636595732/882003276558643271/locks2.jpg
https://cdn.discordapp.com/attachments/630409333636595732/882003274847359027/locks3.jpg
https://cdn.discordapp.com/attachments/630409333636595732/882003273119326308/garage.jpg
https://cdn.discordapp.com/attachments/630409333636595732/882003271844245584/repair_bay.jpg

**Reason:**

I believe that by making this an actual business it would all provide many things to legacy that would be benefitted for months to come. 

Becoming a business would allow for lock to be placed on the doors which would provide a much needed benefit to prevent locals and Danny's from driving through the back entrance. Having door locks on the front doors would also work as a preventative measure as well and if the employees or myself needed to leave for a quick moment, they could lock it up right quick and it would allow for none of the cars to get stolen and the notes to get messed up. The locks would also allow for private conversations to be had in the office without the interruption of random people coming through to talk. With this it would be easier to seal off the room to prevent any sound from escaping to keep conversations that are private, private. 

The garage would allow myself, my future employees, and future owners and their employees to be able to store the cars right in the shop and not have to worry about them being stolen by locals, or worry about having to run back and forth every morning to the nearest house. This garage would also help with performance in and around the area. If the shop has to be suddenly closed then there are  not a influx of cars lying around causing server performance issues with all of the other interiors in the area. Eventually if this was a shared garage it would allow for any employee to sell a vehicle for another employee and allow both of them to receive their share of the vehicle. 

The repair zone in the back would allow for so many unique interactions from all across the city. It would actively engage RP from LA to the mechanic shops in the city as there would not be a mechanic role for LA. Thus requiring the owner and employees to plan for mechanics to be there on certain days to be able to fix up cars in the shop and as well any car that needed a quick fix on the way by. Even it was just a work bench it would also provide great interactions as well because if you currently wanted to get a window, door, or tire put back on you would have to drive halfway across the city to get a repair. Overall the benefit that this would have for LA would be amazing because currently you have to run all of the cars down to a shop in the morning, hoping that Hayes or Lost are open.(which currently at the time that we open LA there are not) Then by the time that you get all of the cars repaired and ready to go you have to go back to the issue of locals stealing your vehicles.

The last request would be for the GPS icon to be added back to the map. This would allow for player both old and new to find the shop easily and it would allow for more business. 

I think all of the changes would provide great additions to Legacy

**Comment by XwezioGH** [comment]
@InZidiuZ I believe this is all script.

**Comment by InZidiuZ** [comment]
Features like these (door locks, for example) are added dynamically through the interior, so it's bound to the map, @XwezioGH. I can add door locks and put them under the 'Mechanic' job as well as the repair spots, though I'm not sure about the garage.

**Comment by InZidiuZ** [comment]
There isn't really any official 'framework business' system though, and if there were, you would've had to redirect that request to the other issue tracker.

**Comment by claymstehling** [comment]
@InZidiuZ Oh okay that makes sense.

My thought with the way the garage would work is that it would be like the Aviation "garages" we currently have. You would have to have that certain role of LA employee, to be able to pull vehicles out. I am not sure if that is possible though. If it isn't I completely understand :D

**Comment by claymstehling** [comment]
> There isn't really any official 'framework business' system though, and if there were, you would've had to redirect that request to the other issue tracker.

And oh okay that makes sense as well I wasn't sure how the business framework looked

**Comment by InZidiuZ** [comment]
I've added the door locks. For the garage, I'm a bit confused as to what you mean. There aren't any garages tied to any jobs. We have the aircraft spawners bound to some state licenses, but that's not a garage.

**Comment by antihero6713** [comment]
the way his car dealership works right now is he has to go to the closest garage or a house to pull out all cars on to the show room floor. He is asking for a garage to be setup in or rear ally he can access the cars quicker.  Currently takes him a lot of time to setup cars and store cars at the business

 He just wants to limit it to him using it or employees only if possible I believe.  which would help if it was positioned inside his place where he could lock it up.

**Comment by InZidiuZ** [comment]
Would this be a good spot for a garage?

![b42d15795f27faa015eaa1c85a25c17d-jpg](https://user-images.githubusercontent.com/22295402/132353378-84f6d6b4-1941-4d83-b850-42fea741bb50.jpg)

**Comment by claymstehling** [comment]
@InZidiuZ Yes! I think that would benefit not only LA but the entire city! :D That is also close enough that it wouldn't despawn the vehicles when running back and forth

**Comment by antihero6713** [comment]
art has been done for this giving it @InZidiuZ  to finish

**Comment by DaxterPancake** [comment]
Is this still relevant?

**Comment by antihero6713** [comment]
nah
